### PR TITLE
Add an appendix describing spin negotiation

### PR DESCRIPTION
**Pull request by britram**
There's been some discussion on the list about using version negotiation to negotiate the spin bit. This PR fleshes that idea out in an appendix, as a basis for discussion in Bangkok.

### CODE REVIEW

**Comment by marten-seemann** [comment]
While I like the idea of exercising version negotiation right from the beginning, the problem with this approach is that this information is only visible on the path that the handshake is performed on. Middleboxes performing RTT measurements will also need to deploy some heuristics for detecting spin-bit enabled flows that migrated.
Having two different ways of detecting the same fact might be error-prone, and maybe it's a better design to force middleboxes to use a single method (which would be heuristics, in this case).

**Comment by kazuho** [comment]
The primary issue with using heuristics is that it leads to the bits getting ossified. You would no longer have a chance to introduce an improved version of "spin bits" without the fear of it being misdetected by the on-path devices that understand the old version.

In practice, I believe that operators do not need to observe all the QUIC connections. Assuming that is true, they can concentrate on QUIC connections that they know that are spinning bits by simply referring to the version. There would be no need for guessing.

File: draft-ietf-quic-spin-exp.md [hunk 1]
```diff
@@ -285,3 +285,59 @@ grant agreement no. 688421 Measurement and Architecture for a Middleboxed
...
+- It provides a method to exercise version negotiation beyond the version
+  greasing mechanism described in section 15 of {{QUIC-TRANSPORT}}. This
+  additional exercise has the benefit of actually changing the protocol's
+  behavior in a way each endpoint can detect based on the negotiation of the
+  version, and may defend against ossification of the version negotiation
+  mechanism caused both by on-path interference as well as lazy implementations
+  better than greasing alone.
+
+- It provides a method by which on-path devices that observe the handshake can
+  explicitly classify a connection as spinning or not.
```
[the remark this answers is not in the document]
Comment (britram) [reply draft-ietf-quic-spin-exp.md]: I think "that observe the handshake" does that here, no?

**Comment by britram** [comment]
@martinthomson yep plan is to merge this and #1931 then point the list at editor's copy for the discussion in BKK.

**Comment by britram** [comment]
@marten-seemann yeah, the benefits of "explicit signaling" are not quite so cut-and-dried. Heuristics for discriminating spinning from non-spinning endpoints are necessary anyway, or rather, the heuristics necessary to reject bad samples due to lost, reordered, or delayed edges will tend to reject non-spinning endpoints as a side effect.

A handshake negotiating version Vs is not a guarantee to the path that endpoints will spin, and in many cases the endpoint will never see a handshake for a 6-tuple due to CID change or migration. However, a handshake negotiating version Vt is a pretty good sign that the observer can ignore the flow, which is a nice optimization.

File: draft-ietf-quic-spin-exp.md [hunk 2]
```diff
@@ -285,3 +285,59 @@ grant agreement no. 688421 Measurement and Architecture for a Middleboxed
...
+
+## The Spin Bit in Spinless QUIC {#nospin}
+
+When version Vt is negotiated, both client and server randomly select a "pin
```
Comment (mirjak) [inline draft-ietf-quic-spin-exp.md]: "pin" or "spin"...?

**Pull request merged**